components/threatconnect.js

```diff
@@ -87,12 +103,261 @@ polarity.export = PolarityComponent.extend({
               default:
                 caseObj.__severityColor = '';
             }
+
+            switch (caseObj.status) {
+              case 'Open':
+                caseObj.__statusColor = 'green-text';
+                break;
+              case 'Closed':
+                caseObj.__statusColor = 'red-text';
+                break;
+              default:
+                caseObj.__statusColor = '';
+            }
           });
         }
       }
     }
   },
   actions: {
+    toggleEdit(caseId, indicatorId) {
+      const indicatorPath = `indicators.${indicatorId}.indicator.associatedCases.data`;
+      const casesArray = this.get(indicatorPath);
+      const caseToEdit = casesArray.find((c) => c.id === caseId);
+
+      if (!caseToEdit) return;
+
+      const caseIndex = casesArray.indexOf(caseToEdit);
+
+      const isEditing = this.get(`${indicatorPath}.${caseIndex}.__isEditing`) || false;
+
+      if (isEditing) {
+        this.send('saveCaseUpdates', caseId, indicatorId);
+      } else {
+        this.set(`${indicatorPath}.${caseIndex}.__isEditing`, true);
+      }
+    },
+    cancelEdit(caseId, indicatorId) {
+      const indicatorPath = `indicators.${indicatorId}.indicator.associatedCases.data`;
+      const casesArray = this.get(indicatorPath);
+      const caseToEdit = casesArray.find((c) => c.id === caseId);
+
+      if (!caseToEdit) return;
+
+      const caseIndex = casesArray.indexOf(caseToEdit);
+
+      this.set(`${indicatorPath}.${caseIndex}.__isEditing`, false);
+    },
+    saveCaseUpdates(caseId, indicatorId) {
+      const indicatorPath = `indicators.${indicatorId}.indicator.associatedCases.data`;
+      const casesArray = this.get(indicatorPath);
+      const caseToUpdate = casesArray.find((c) => c.id === caseId);
+
+      if (!caseToUpdate) return;
+
+      const caseIndex = casesArray.indexOf(caseToUpdate);
+
+      this.set(`${indicatorPath}.${caseIndex}.__updating`, true);
+      this.set(`${indicatorPath}.${caseIndex}.__isSaving`, true);
+
+      const newValues = {
+        status: caseToUpdate.__newStatus,
+        severity: caseToUpdate.__newSeverity,
+        resolution: caseToUpdate.__newResolution,
+        description: caseToUpdate.__newDescription,
+        attributes: { data: caseToUpdate.__newAttributes }
+      };
+
+      const filteredValues = {};
+      Object.entries(newValues).forEach(([key, value]) => {
+        if (value) {
+          filteredValues[key] = value;
+        }
+      });
+
+      const payload = {
+        action: 'UPDATE_CASE',
+        caseId,
+        mode: 'append'
+      };
+
+      Object.assign(payload, filteredValues);
+
+      const shouldUpdate = Object.values(newValues).some((value) => value);
+
+      if (shouldUpdate) {
+        this.sendIntegrationMessage(payload)
+          .then((result) => {
```

Comment:
After successfully editing a case, we need to reapply the "status" and "severity" colors.

I would recommend refactoring the current logic that applies the colors out of the `onDetailsLoaded` method and making it a utility function.  Something like:

```
applyStatusColorToCase(case){
      switch (case.status) {
        case 'Open':
          case.__statusColor = 'green-text';
          break;
        case 'Closed':
          case.__statusColor = 'red-text';
          break;
        default:
          case.__statusColor = '';
      }
}
```

And do the same thing for severity.  Then just call those methods anywhere we need to re-apply the colors (can be done in onDetails, as well as the `saveCaseUpdates` and probably also needed in the method that creates new cases.